### PR TITLE
Add some architecture diagrams

### PR DESCRIPTION
**Pull request by phoban01**
## Description

This PR adds some basic architecture diagrams to get us started. 

Draw.io source files also included.

### CODE REVIEW

**Review by Skarlso — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,6 +4,18 @@
 
 Main repository for ocm-controller.
 
+## Architecture
+
+The `ocm-controller` manages multiple Custom Resources and therefore contains several reconcilers. It also provides an OCI registry used to cache blobs and make them accessible to clients within the cluster. 
+
+![ocm controller reconcilers](./docs/diagrams/ocm-controller-reconcilers.png)
+
+The `ocm-controller` is responsible for the core work necessary to utilise resources from an `OCM` component in a Kubernetes cluster. This includes resolving `ComponentDescriptor` metadata for a particular component version, performing authentication to OCM repositories, retrieving artifacts from OCM repositories, making individual resources from the OCM component available within the cluster, performing localization and configuration.
```
Comment (nikimanoledaki) [inline README.md:13]:
```suggestion
The `ocm-controller` is responsible for the core work necessary to utilise resources from an `OCM` component in a Kubernetes cluster. This includes:
- Resolving [`ComponentDescriptor`](https://github.com/open-component-model/ocm-spec/blob/main/doc/glossary.md#component-descriptor) metadata for a particular [component version](https://github.com/open-component-model/ocm-spec/blob/main/doc/glossary.md#component-version)
- Performing authentication with [OCM repositories](https://github.com/open-component-model/ocm-spec/blob/main/doc/glossary.md#component-repository)
- Retrieving [artifacts](https://github.com/open-component-model/ocm-spec/blob/main/doc/glossary.md#artifact) from OCM repositories
- Making individual resources from the OCM component available within the cluster
- Performing localization and configuration
```
This makes it a bit easier to read/skim :)

Comment (nikimanoledaki) [reply README.md:13]: > localization and configuration

Are these explained further somewhere that we can link to?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,18 @@
...
 
+## Architecture
+
+The `ocm-controller` manages multiple Custom Resources and therefore contains several reconcilers. It also provides an OCI registry used to cache blobs and make them accessible to clients within the cluster. 
+
+![ocm controller reconcilers](./docs/diagrams/ocm-controller-reconcilers.png)
+
+The `ocm-controller` is responsible for the core work necessary to utilise resources from an `OCM` component in a Kubernetes cluster. This includes resolving `ComponentDescriptor` metadata for a particular component version, performing authentication to OCM repositories, retrieving artifacts from OCM repositories, making individual resources from the OCM component available within the cluster, performing localization and configuration. 
+
+Snapshots are used to pass blobs between controllers and snapshots are stored in the internal registry managed by the `ocm-controller`.
```
Comment (nikimanoledaki) [inline README.md:15]: > Snapshots

Is this explained further somewhere that we can link to?

**Review by nikimanoledaki — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (phoban01) [reply README.md:20]: Not currently.

**Review by nikimanoledaki — APPROVED** [review]

**Pull request merged**